specs/core/0_beacon-chain.md

```diff
@@ -1632,19 +1623,17 @@ def process_final_updates(state: BeaconState) -> None:
     state.current_epoch_attestations = []
 ```
 
-### Per-slot processing
-
-At every `slot > GENESIS_SLOT` run the following function:
+### Block processing
 
 ```python
-def advance_slot(state: BeaconState) -> None:
-    state.slot += 1
+def process_block(state: BeaconState, block: BeaconBlock) -> None:
+    process_block_header(state, block)
+    process_randao(state, block)
+    process_eth1_data(state, block)
+    process_operations(state, block.body)
+    # verify_block_state_root(state, block)
```

Comment:
I'd say enable it in the spec, and hack it in the tests.

/cc @djrtwo

Reply:
a bit torn here. 

options:
1. uncomment and leave as is
2. add `verify_state_root` boolean in `process_block` and `state_transition` and a note about in which context you would run it as True or False (receiving a block vs building a block)
3. have `verify_block_state_root` live outside of the state transition function entirely and as a note on block validity below.

I think I like (2) the best, but am open to debate.
*

Reply:
My suggestion would be for the test bench to populate the `state_root` immediately prior to verification in case it is set to `ZERO_HASH`. That way we can keep the spec clean.

Reply:
some sort of monkey patch?

Reply:
Yes